tests/objects/test_operation.py

```diff
@@ -215,7 +216,9 @@ def test_event_logs(self, loop, op_for_event_logs, operation_agent, file_svc, da
         assert event_logs == want
 
     def test_writing_event_logs_to_disk(self, loop, op_for_event_logs, operation_agent, file_svc, data_svc):
+        loop.run_until_complete(data_svc.remove('agents', match=dict(unique=operation_agent.unique)))
```

Comment:
I lost the github action run (probably because I force pushed some commits after rebasing) but this test was failing. After poking around I realized that storing the agent via the data_svc on the next line is basically a no-op if it already exists: https://github.com/mitre/caldera/blob/bb31c3304f39454e81d5b97083e832f8c200d3aa/app/objects/c_agent.py#L119-L124

The test was failing because of a 1 second time difference on the created timestamp of the stored agent metadata and the expected agent metadata. This is because the expected agent metadata was built from the fixture agent, while the _found_ agent was the previously stored one.

There are other ways to address this but this was quick and got the job done. I think we need to go through our tests and identify all the scope leaks and fix them.

Reply:
@uruwhy fyi

Reply:
Ah, thanks for that!